pkg/osquery/runtime/runner.go

```diff
@@ -49,6 +53,31 @@ func New(k types.Knapsack, serviceClient service.KolideService, opts ...OsqueryI
 }
 
 func (r *Runner) Run() error {
+	for {
+		// if our instances ever exit unexpectedly, return immediately
```

Comment:
I think this comment comment isn't accurate? I think `runRegisteredInstances` only returns if a shutdown was requested. And it only returns an error if a) shutdown was requested and b) we were trying to restart one or more instances during that time and c) hadn't successfully restarted one of them yet.

Either way, why would we return the error here instead of checking `rerunRequired` first?

Reply:
that's a good point thank you. we'd probably want to rerun regardless if required, I will update that comment and get this fixed up!